dashboard/test/models/script_test.rb

```diff
@@ -194,6 +194,52 @@ def populate_cache_and_disconnect_db
     assert_equal 'Stage1', second.name
   end
 
+  test 'all fields can be set and then removed on reseed' do
+    # First, seed using a .script file that sets something truthy for all properties.
+    script_file_all_properties = File.join(self.class.fixture_path, 'test-all-fields.script')
+    scripts, _ = Script.setup([script_file_all_properties])
+    script = scripts.first
+
+    # Not testing new_name since it causes a new script to be created.
+    assert_equal false, script.hidden? # defaults to true, so we want to verify it was explicitly set to false
+    assert script.login_required?
+    assert_equal 'csd1', script.family_name
+
+    # Seed using an empty .script file with the same name. Verify that this sets all values back to defaults.
+    script_file_no_properties = File.join(self.class.fixture_path, 'duplicate_scripts', 'test-all-fields.script')
+    Script.setup([script_file_no_properties])
+    script.reload
+
+    assert script.hidden? # defaults to true
+    assert_equal false, script.login_required?
+    assert_nil script.family_name
+  end
+
+  test 'all properties can be set and then removed on reseed' do
+    # First, seed using a .script file that sets something truthy for all properties.
+    script_file_all_properties = File.join(self.class.fixture_path, 'test-all-properties.script')
+    scripts, _ = Script.setup([script_file_all_properties])
+    script = scripts.first
+
+    assert_equal 20, script.properties.keys.length
+    script.properties.values.each {|v| assert v}
+
+    # Seed using an empty .script file with the same name. Verify that this sets all values back to defaults.
+    script_file_no_properties = File.join(self.class.fixture_path, 'duplicate_scripts', 'test-all-properties.script')
+    Script.setup([script_file_no_properties])
+    script.reload
+
+    # The values below do not get reset to defaults. TODO: determine if this behavior should be changed.
+    expected_props = {
+      "is_stable" => true,
+      "peer_reviews_to_complete" => 1,
```

Comment:
@islemaster do you think it is safe to clear the `peer_reviews_to_complete` field on scripts in production, for all scripts which no longer have this field set in their script file? Of those script files containing `professional_learning_course`, I see a mix of whether this field is present or not, and I don't have a sense of how to spot a case where the `peer_reviews_to_complete` field might be missing.

Reply:
I checked production. There are currently 45 scripts on production that have a nonzero `peer_reviews_to_complete` value. 44 of them match up exactly with what I'm seeing in our script files, so this should be a safe change for those.

The only script on production that does not have the expected `peer_reviews_to_complete` value in its script file is `csp-online-test`, which was created back in September 2016 and only has 6 UserScripts.  I don't think we need to worry about it.